lib/cryptoexchange/exchanges/bitholic/services/market.rb

```diff
@@ -0,0 +1,48 @@
+module Cryptoexchange::Exchanges
+  module Bitholic
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super(ticker_url)
+          adapt_all(output)
+        end
+
+        def adapt_all(output)
+          output['market'].map do |pair, payload|
+            target, base = pair.split('_')
+            market_pair = Cryptoexchange::Models::MarketPair.new(
+              base:   base,
+              target: target,
+              market: Bitholic::Market::NAME
+            )
+            adapt(payload, market_pair)
+          end
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::Bitholic::Market::API_URL}/ticker"
+        end
+
+        def adapt(payload, market_pair)
+          ticker           = Cryptoexchange::Models::Ticker.new
+          ticker.base      = market_pair.base
+          ticker.target    = market_pair.target
+          ticker.market    = Bitholic::Market::NAME
+          ticker.ask       = NumericHelper.to_d(payload['ask'])
+          ticker.bid       = NumericHelper.to_d(payload['bid'])
+          ticker.last      = NumericHelper.to_d(payload['last'])
+          ticker.volume    = NumericHelper.to_d(payload['volume24h'])
```

Comment:
Volume here has to be in base currency. eg. QTUM/BTC, volume should be in QTUM
But volume24h is returning volume in BTC. Refer to other exchange implementation that uses the `divide` method helper to calculate the volume in base

Reply:
I understand that you need the volume generated based on base currency such as BTC and QTUM.
For example, in the ETH/BTC trading pairs, we are currently generating the trading volume based on ETH.
But you mean that in the ETH/BTC trading pairs, we would have to make the trading volume based on BTC, right?

If it's right, I will have my dev team do this.

Reply:
FYI, ETH/BTC means ETH is base currency and BTC is the market.

Reply:
You and I seem to have a different concept of A/B structure. I might know. after discussion with dev, I will let you know